src/ui/views/session/create-world/ScenePreviewUpload.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState, useCallback } from "react";
+import { IBlobHandle } from "@thirdroom/hydrogen-view-sdk";
+
+import { useHydrogen } from "../../../hooks/useHydrogen";
+import { useAttachmentUpload } from "../../../hooks/useAttachmentUpload";
+import { Label } from "../../../atoms/text/Label";
+import { Text } from "../../../atoms/text/Text";
+import { Button } from "../../../atoms/button/Button";
+import { Icon } from "../../../atoms/icon/Icon";
+import { SettingTile } from "../../components/setting-tile/SettingTile";
+import { bytesToSize } from "../../../utils/common";
+import { useDebounce } from "../../../hooks/useDebounce";
+import UploadIC from "../../../../../res/ic/upload.svg";
+
+export function ScenePreviewUpload({
+  onMxcChange,
+  onBlobChange,
+}: {
+  onMxcChange: (url?: string) => void;
+  onBlobChange: (blob?: IBlobHandle) => void;
+}) {
+  const { session, platform } = useHydrogen(true);
+  const [scenePrevBlob, setScenePrevBlob] = useState<IBlobHandle>();
+  const [sentBytes, setProgress] = useState(0);
+  const handleProgress = useCallback((sentBytes: number) => {
```

Comment:
This `useCallback` is unnecessary. You can use `setProgress` directly instead of `handleProgress`